packages/core/src/util/transactionSummaryInspector.ts

```diff
@@ -0,0 +1,242 @@
+import * as Cardano from '../Cardano';
+import { AssetId } from '../Cardano';
+import {
+  AssetsMintedInspection,
+  Inspector,
+  ResolutionResult,
+  assetsBurnedInspector,
+  assetsMintedInspector,
+  resolveInputs,
+  totalAddressInputsValueInspector,
+  totalAddressOutputsValueInspector
+} from './txInspector';
+import { BigIntMath } from '@cardano-sdk/util';
+import { addTokenMaps, subtractTokenMaps } from '../Asset/util';
+import { coalesceValueQuantities } from './coalesceValueQuantities';
+import { subtractValueQuantities } from './subtractValueQuantities';
+
+interface TransactionSummaryInspectorArgs {
+  addresses: Cardano.PaymentAddress[];
+  rewardAccounts: Cardano.RewardAccount[];
+  inputResolver: Cardano.InputResolver;
+  protocolParameters: Cardano.ProtocolParameters;
+}
+
+export type TransactionSummaryInspection = {
+  tokens: Cardano.TokenMap;
+  coins: Cardano.Lovelace;
+  collateral: Cardano.Lovelace;
+  deposit: Cardano.Lovelace;
+  returnDeposit: Cardano.Lovelace;
+  totalFees: Cardano.Lovelace;
+  unresolved: {
+    inputs: Cardano.TxIn[];
+    value: Cardano.Value;
+  };
+};
+
+export type TransactionSummaryInspector = (
+  args: TransactionSummaryInspectorArgs
+) => Inspector<TransactionSummaryInspection>;
+
+type DepositProtocolParams = { stakeKeyDeposit: Cardano.Lovelace; poolDeposit: Cardano.Lovelace };
+
+const computeShellyDeposits = (
+  depositParams: DepositProtocolParams,
+  certificates: Cardano.Certificate[],
+  rewardAccounts: Cardano.RewardAccount[],
+  poolIds: Set<Cardano.PoolId>,
+  networkId: Cardano.NetworkId
+): { deposit: Cardano.Lovelace; returnedDeposit: Cardano.Lovelace } => {
+  let deposit = 0n;
+  let returnedDeposit = 0n;
+
+  // TODO: For the case of deregistration (StakeDeregistration and PoolRetirement) the code here is not entirely correct
```

Comment:
Let's capture this in a ticket. Maybe @mirceahasegan can advice an appropriate parent ticket

Reply:
done: LW-9604